app/controllers/ht_approval_requests_controller.rb

```diff
@@ -3,18 +3,26 @@
 class HTApprovalRequestsController < ApplicationController
   before_action :fetch_requests, only: %i[show update edit]
 
-  def create
-    return unless params[:submit_req]
-
-    adds = add_requests(params[:ht_users])
-    flash[:notice] = "Added #{'request'.pluralize(adds.count)} for #{adds.join ', '}" if adds.count.positive?
+  def create # rubocop:disable Metrics/MethodLength
+    if params[:submit_requests]
+      @added_users = add_requests(params[:ht_users])
+      flash[:notice] = "Added #{'request'.pluralize(@added_users.count)} for #{@added_users.join ', '}" if @added_users.count.positive?
+      session[:added_users] = @added_users
+    end
+    if params[:submit_renewals]
+      @renewed_users = add_renewals(params[:ht_users])
+      flash[:notice] = "Renewed #{@renewed_users.join ', '}" if @renewed_users.count.positive?
+      session[:renewed_users] = @renewed_users
+    end
     redirect_to action: :index
-    session[:added_users] = adds
   end
 
   def index
     @reqs = HTApprovalRequest.order('approver')
     @added_users = session[:added_users] || []
+    @renewed_users = session[:renewed_users] || []
+    session.delete :added_users
```

Comment:
Looks like this was the fix for the stickiness?

Reply:
Yes, it was. I initially tried storing it in the flash, but the way our header deals with flash produced unwanted artifacts, so I went with this (I think flash is part of the session anyway, so six/half dozen).